api/types/container/host_config.go

```diff
@@ -311,6 +312,9 @@ type Resources struct {
 	Memory    int64 // Memory limit (in bytes)
 	NanoCPUs  int64 `json:"NanoCpus"` // CPU quota in units of 10<sup>-9</sup> CPUs.
 
+	// Applicable to all platforms, but not guaranteed to be supported.
+	GPUsByVendor map[gpu.Vendor]*gpu.Resources `json:"GpusByVendor,omitempty"` // List of GPUs to map inside the container.
```

Comment:
This limits the API quite a bit.
What do you think about using a slice here and then we can validate as needed in code.